test/db/postgresql/oid_types_test.rb

```diff
@@ -40,31 +34,21 @@ class SomeSample < ActiveRecord::Base
   end
 
   def test_resolves_oid_type
-    column = SomeSample.columns_hash['str']
-    if ar_version('4.2')
-      assert_instance_of ActiveRecord::Type::String, column.cast_type
-    else # 4.1/4.0
-      assert_kind_of OID::Identity, column.oid_type
-    end
+    assert_instance_of ActiveRecord::Type::String, SomeSample.type_for_attribute('str')
   end
 
   def test_returns_column_and_resolves_oid_type
     adapter = ActiveRecord::Base.connection
-    if defined? JRUBY_VERSION
-      column = adapter.column_for('some_samples', :int)
-    else
-      column = adapter.columns('some_samples').find { |c| c.name.to_sym == :int }
-    end
+    column = adapter.column_for('some_samples', :int)
     assert_not_nil column
-    assert_instance_of OID::Integer, oid_type(column)
-  end if ar_version('4.1')
+    assert_instance_of ActiveRecord::Type::Integer, column.type
+  end
 
   def test_returns_column_accessor_for_hstore
```

Comment:
@enebo Sound good. I have another PR about ready with some more fixes so I'll remove it in that one.